### PR TITLE
Update account API to include derivedParties

### LINKED ISSUE #11265: [vAMM] update account API to include derivedParties
update the account APIs to include derivedParties (see position API for reference)

### PR DESCRIPTION
**Pull request by karlem**
Close #11265

### CODE REVIEW

File: protos/sources/data-node/api/v2/trading_data.proto [hunk 1]
```diff
@@ -1181,6 +1187,9 @@ message ObserveAccountsRequest {
   string asset = 3;
   // Account type to subscribe to, required field.
   vega.AccountType type = 4;
+  // Whether to return all derived parties from AMMs for the given party.
+  // If used, party ID is required.
+  optional bool include_derived_parties = 5;
```
Comment (candida-d) [inline protos/sources/data-node/api/v2/trading_data.proto:1192]: These new fields probably also need to be included in GraphQL

Comment (karlem) [reply protos/sources/data-node/api/v2/trading_data.proto:1192]: You are right. Just added it.

Comment (candida-d) [reply protos/sources/data-node/api/v2/trading_data.proto:1192]: @karlem Should the parent party ID field go in too, or is it already there? I didn't see it in the changes.

Comment (karlem) [reply protos/sources/data-node/api/v2/trading_data.proto:1192]: Maybe not. let me double check

Comment (karlem) [reply protos/sources/data-node/api/v2/trading_data.proto:1192]: Should be theere.

**Review by candida-d — DISMISSED** [review]

File: protos/sources/data-node/api/v2/trading_data.proto [hunk 2]
```diff
@@ -1139,6 +1139,9 @@ message AccountBalance {
   string market_id = 5;
   // Account type of this account.
   vega.AccountType type = 6;
+  // Parent party ID of the account.
+  // This field is optional and is used in cases where the account is owned by a derived key.
```
Comment (wwestgarth) [inline protos/sources/data-node/api/v2/trading_data.proto:1143]:
```suggestion
  // This field is optional and is used in cases where the account is derived from another party's account.
```

**Review by wwestgarth — DISMISSED** [review]

**Review by wwestgarth — APPROVED** [review]

**Pull request merged**